src/server/generic_family.cc

```diff
@@ -463,6 +458,164 @@ void GenericFamily::Stick(CmdArgList args, ConnectionContext* cntx) {
   (*cntx)->SendLong(match_cnt);
 }
 
+struct SortEntryEmpty {};
+
+struct SortEntryScore {
+  double score;
+};
+
+template <
+    // Whether alphanumeric sort will be used.
+    bool ALPHA,
+    // Deduced StorageKind type for optimal size.
+    container_utils::StorageKind STORAGE_KIND,
+    // Whether to store double score. It will be stored for non-alphanumeric sorts
+    // when our items are not trivailly convertiable to doubles (i.e. they are not integers)
+    bool STORE_SCORE = !ALPHA && STORAGE_KIND != container_utils::StorageKind::INTEGRAL>
+struct SortEntry : public std::conditional_t<STORE_SCORE, SortEntryScore, SortEntryEmpty> {
+  // Key type. If we sort alphanumerically, we store strings. Else we mimic
+  // the container storage type.
+  using KEY_TYPE =
+      std::conditional_t<ALPHA, std::string, container_utils::StorageType<STORAGE_KIND> >;
+  // Comparison type. std:string for alphanumeric sort, double otherwise.
+  using COMPARE_TYPE =
+      std::conditional_t<ALPHA, const std::string&,
+                         // If we store only integers, we can use them for comparison.
+                         std::conditional_t<std::is_integral_v<KEY_TYPE>, KEY_TYPE, double> >;
+
+  // This is either: std::string, int64_t or std::variant<std::string, int64_t>
+  KEY_TYPE key;
+
+  bool Parse(std::string&& item) {
+    if constexpr (STORE_SCORE) {
+      if (!absl::SimpleAtod(item, &this->score))
+        return false;
+    }
+    if constexpr (std::is_assignable_v<KEY_TYPE&, std::string&&>) {
+      key = std::move(item);
+    } else {
+      assert(false && "unreachable");
+    }
+    return true;
+  }
+
+  bool Parse(int64_t item) {
+    if constexpr (STORE_SCORE) {
+      this->score = item;
+    }
+    if constexpr (std::is_assignable_v<KEY_TYPE&, int64_t>) {
+      key = item;
+    } else {
+      assert(false && "unreachable");
+    }
+    return true;
+  }
+
+  COMPARE_TYPE Cmp() const {
+    if constexpr (STORE_SCORE) {
+      return this->score;
+    } else {
+      return key;
+    }
+  }
+
+  std::string DEBUG_Format() const {
+    using namespace container_utils;
+    StorageType<StorageKind::MIXED> mixed = key;
+    return std::visit(
+        [](const auto& val) -> std::string {
+          if constexpr (std::is_same_v<std::decay_t<decltype(val)>, std::string>)
+            return val;
+          else
+            return std::to_string(val);
+        },
+        mixed);
+  }
+};
+
+template <bool ALPHA, container_utils::StorageKind STORAGE_KIND>
+OpStatus SortInternal(PrimeIterator it) {
+  using SENTRY = SortEntry<ALPHA, STORAGE_KIND>;
+  std::vector<SENTRY> sort_entries;
+
+  VLOG(0) << "Storage kind " << int(STORAGE_KIND) << " size " << sizeof(SENTRY);
```

Comment:
`DVLOG(1)` probably

Reply:
I'll add normal logs once this is no longer a prototype. This is just for me so far